pkg/netceptor/firewall_rules_test.go

```diff
@@ -83,3 +97,436 @@ func TestFirewallRules(t *testing.T) {
 		t.Fatal("rule #4 did not return Reject")
 	}
 }
+
+func TestParseFirewallRules(t *testing.T) {
+	tests := []struct {
+		name            string
+		rules           []FirewallRuleData
+		wantErr         bool
+		wantErrContains string
+		wantCount       int
+		testCases       []firewallRuleTestCase
+	}{
+		// Test with empty rules slice
+		{
+			name:      "empty rules slice",
+			rules:     []FirewallRuleData{},
+			wantErr:   false,
+			wantCount: 0,
+		},
+		// Test with single valid rule
+		{
+			name: "single valid rule",
+			rules: []FirewallRuleData{
+				{
+					"action":   "accept",
+					"fromnode": "node1",
+				},
+			},
+			wantErr:   false,
+			wantCount: 1,
+			testCases: []firewallRuleTestCase{
+				{
+					desc:      "rule 0 accepts matching node1",
+					ruleIndex: 0,
+					data:      &MessageData{FromNode: "node1"},
+					want:      FirewallResultAccept,
+				},
+				{
+					desc:      "rule 0 continues for non-matching node2",
+					ruleIndex: 0,
+					data:      &MessageData{FromNode: "node2"},
+					want:      FirewallResultContinue,
+				},
+			},
+		},
+		// Test with multiple valid rules
+		{
+			name: "multiple valid rules",
+			rules: []FirewallRuleData{
+				{
+					"action":   "accept",
+					"fromnode": "node1",
+				},
+				{
+					"action":   "reject",
+					"fromnode": "node2",
+				},
+			},
+			wantErr:   false,
+			wantCount: 2,
+			testCases: []firewallRuleTestCase{
+				{
+					desc:      "rule 0 accepts matching node1",
+					ruleIndex: 0,
+					data:      &MessageData{FromNode: "node1"},
+					want:      FirewallResultAccept,
+				},
+				{
+					desc:      "rule 0 continues for non-matching node2",
+					ruleIndex: 0,
+					data:      &MessageData{FromNode: "node2"},
+					want:      FirewallResultContinue,
+				},
+				{
+					desc:      "rule 1 rejects matching node2",
+					ruleIndex: 1,
+					data:      &MessageData{FromNode: "node2"},
+					want:      FirewallResultReject,
+				},
+				{
+					desc:      "rule 1 continues for non-matching node1",
+					ruleIndex: 1,
+					data:      &MessageData{FromNode: "node1"},
+					want:      FirewallResultContinue,
+				},
+			},
+		},
+		// Test wiith invalid rule data
+		{
+			name: "invalid rule",
+			rules: []FirewallRuleData{
+				{
+					"action": "",
+				},
+			},
+			wantErr:   true,
+			wantCount: 0,
```

Comment:
Would this make sense?
```suggestion
			wantCount: 0,
			wantErrContains: "unknown action:",
```